example/lib/values.g.dart

```diff
@@ -306,6 +308,67 @@ class _$WireNameValueSerializer implements StructuredSerializer<WireNameValue> {
   }
 }
 
+class _$MigratedWireNameValueSerializer
+    implements StructuredSerializer<MigratedWireNameValue> {
+  @override
+  final Iterable<Type> types = const [
+    MigratedWireNameValue,
+    _$MigratedWireNameValue
+  ];
+  @override
+  final String wireName = 'V';
+
+  @override
+  Iterable<Object> serialize(
+      Serializers serializers, MigratedWireNameValue object,
+      {FullType specifiedType = FullType.unspecified}) {
+    final result = <Object>[
+      'value',
+      serializers.serialize(object.value, specifiedType: const FullType(int)),
+      'validatedValue',
+      serializers.serialize(object.validatedValue,
+          specifiedType: const FullType(ValidatedValue)),
+    ];
+
+    return result;
+  }
+
+  @override
+  MigratedWireNameValue deserialize(
+      Serializers serializers, Iterable<Object> serialized,
+      {FullType specifiedType = FullType.unspecified}) {
+    final result = new MigratedWireNameValueBuilder();
+
+    final iterator = serialized.iterator;
+    while (iterator.moveNext()) {
+      final key = iterator.current as String;
+      iterator.moveNext();
+      final Object value = iterator.current;
+      switch (key) {
+        case 'v':
+          result.value ??= serializers.deserialize(value,
+              specifiedType: const FullType(int)) as int;
+          break;
+        case 'value':
+          result.value = serializers.deserialize(value,
+              specifiedType: const FullType(int)) as int;
+          break;
+        case 'validatedValueV1':
+        case 'validatedValueV2':
+          result.validatedValue.replace(serializers.deserialize(value,
```

Comment:
Oh. Hmm. That's harder than I was hoping.

One way to do it would be to declare a bool e.g. $haveSetValidatedValue above, then split the 'case' into one for the fundamental name and one for everything else. The one for the fundamental name always calls 'replace' and also sets the bool to true, the one for all the alt names checks the bool and only calls 'replace' if it's false.